lxd/storage/drivers/driver_pure_volumes.go

```diff
@@ -1219,14 +1219,107 @@ func (d *pure) DeleteVolumeSnapshot(snapVol Volume, op *operations.Operation) er
 	return nil
 }
 
-// MountVolumeSnapshot simulates mounting a volume snapshot.
+// MountVolumeSnapshot creates a new temporary volume from a volume snapshot to allow mounting it.
 func (d *pure) MountVolumeSnapshot(snapVol Volume, op *operations.Operation) error {
-	return d.MountVolume(snapVol, op)
+	revert := revert.New()
+	defer revert.Fail()
+
+	parentVol := getSnapshotParentVolume(snapVol)
+
+	// Get the parent volume name.
+	parentVolName, err := d.getVolumeName(parentVol)
+	if err != nil {
+		return err
+	}
+
+	// Get the snapshot volume name.
+	snapVolName, err := d.getVolumeName(snapVol)
+	if err != nil {
+		return err
+	}
+
+	// A PureStorage snapshot cannot be mounted. To mount a snapshot, a new volume
+	// has to be created from the snapshot.
+	err = d.client().copyVolumeSnapshot(snapVol.pool, parentVolName, snapVolName, snapVol.pool, snapVolName)
```

Comment:
can this temporary name conflict with other volume names?

Reply:
No. Snapshots are prefixed with `s` which is only the case for snapshots. Similarly, it *should* not conflict with other snapshots as they have a unique volatile UUID.

The only potential case where it could cause the problem: if the same volume with snapshots is being refreshed simultaneously to two different locations, but AFAIK the locks should prevent such case?